csrc/codegen.cpp

```diff
@@ -246,6 +249,24 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
     return val_to_name_.at(v);
   }
 
+  // If the variable is an aligned array, append ".array" to use the reguar
+  // array. This avoid the type mismatch in template functions when one of the
+  // arguments is an aligned array (Array<T,N>) while the other is a regular
+  // array T[N].
+  std::string genVarForTemplateFunction(Val* v) {
```

Comment:
Should this be named something like `maybeConvertAlignedArrayToRawArray`?

Reply:
I rename to `genVariableNameConvertAlignedArray` to reflect its two functions: (1) call `genVariableName` and (2) ConvertAlignedArrayToRawArray